src/function.ts

```diff
@@ -61,28 +112,347 @@ export class Function<P, O> implements Integration {
+            const reversed =
+              Tokenization.reverse(obj, { failConcat: false }) ??
+              Tokenization.reverseString(obj).tokens;
+            if (!Array.isArray(reversed) || reversed.length > 0) {
+              if (Array.isArray(reversed)) {
+                tokens = [...tokens, ...reversed.map((s) => s.toString())];
+              } else {
+                tokens = [...tokens, reversed.toString()];
+              }
+            }
+          }
+          return true;
+        },
+      }
+    );
+
+    const tokenContext = tokens.map((t) => {
+      const id = new RegExp("\\${Token\\[TOKEN\\.([0-9]*)\\]}", "g").exec(
+        t
+      )?.[1];
+      return {
+        id,
+        token: t,
+        // env variables must start with a alpha character
+        env: `env__functionless${id}`,
+      };
+    });
+
+    const resultText = tokenContext.reduce(
+      // TODO: support templated strings
+      (r, t) => r.split(`"${t.token}"`).join(`process.env.${t.env}`),
+      result.text
+    );
+
+    // console.log(resultText);
+
+    const asset = aws_lambda.Code.fromAsset("", {
+      assetHashType: AssetHashType.OUTPUT,
+      bundling: {
+        image: DockerImage.fromRegistry("empty"),
+        // This forces the bundle directory and cache key to be unique. It does nothing else.
+        user: scope.node.addr,
+        local: {
+          tryBundle(outdir: string) {
+            fs.writeFileSync(path.resolve(outdir, "index.js"), resultText);
+            return true;
+          },
+        },
+      },
+    });
+
+    if (!(scope instanceof aws_lambda.Function)) {
+      throw new Error(
+        "CallbackLambdaCode can only be used on aws_lambda.Function"
+      );
+    }
+
+    tokenContext.forEach((t) => scope.addEnvironment(t.env, t.token));
+
+    const funcResource = scope.node.findChild(
+      "Resource"
+    ) as aws_lambda.CfnFunction;
+
+    const codeConfig = asset.bind(scope);
+
+    funcResource.code = {
+      s3Bucket: codeConfig.s3Location?.bucketName,
+      s3Key: codeConfig.s3Location?.objectKey,
+      s3ObjectVersion: codeConfig.s3Location?.objectVersion,
+      zipFile: codeConfig.inlineCode,
+      imageUri: codeConfig.image?.imageUri,
+    };
+
+    asset.bindToResource(funcResource);
+  }
+}
+
+export interface NativeIntegration<F extends AnyFunction> {
+  bootstrap: (context: Function<any, any>) => void;
+  /**
+   * @param args The arguments passed to the integration function by the user.
+   * @param preWarmContext contains singleton instances of client and other objects initialized outside of the native
+   *                       function handler.
+   */
+  call: (
+    args: Parameters<F>,
+    preWarmContext: NativePreWarmContext
+  ) => Promise<ReturnType<F>>;
+  /**
+   * Method called outside of the handler to initialize things like the PreWarmContext
+   */
+  preWarm?: (preWarmContext: NativePreWarmContext) => void;
+}
+
+/* istanbul ignore next */
```

Comment:
why do i keep seeing this istanbul thing everywhere?

Also, this class seems important - let's add docs.

Reply:
Removed them all, just means we might have slightly bigger bundles while testing and that is fine.